src/core/constraints/ShapeBasedConstraint.cpp

```diff
@@ -17,6 +18,30 @@ Vector3d ShapeBasedConstraint::total_force() const {
     return total_force;
 }
 
+
+double ShapeBasedConstraint::mindist() {
```

Comment:
This is parallel code (it runs on all nodes in synchronous mode), so you can't use `partCfg`, as it exists only on the head node. There is also no need, as your algorithm does not need global information, you can run it on each node locally and then reduce the local results to the master. Also folded coordinates should be used with the shape, but particles in partCfg have folded coordinates. So you could do something along the lines of

```c++
auto & parts = local_cells.particles();

auto const local_mindist = std::accumulate(parts.begin(), parts.end(), std::numeric_limits<double>::infinity(), [this](double min, Particle const& p) {
  if(!checkIfInteraction(...))
    return min;
  else  {
     double dummy[3], dist;
     m_shape->calculate_dist(folded_position(p).data(), &dist, dummy); 

     return std::min(min, dist);
   });

   double global_min;
   boost::mpi::reduce(comm_cart, local_min, global_min, std::min, 0);

   return global_min;
}
```  
}